pandas/core/indexing.py

```diff
@@ -944,6 +944,9 @@ def _handle_lowerdim_multi_index_axis0(self, tup):
         except TypeError:
             # slices are unhashable
             pass
+        except KeyError as ek:
+            if len(tup) > self.obj.ndim:
+                raise ek
         except Exception as e1:
             if isinstance(tup[0], (slice, Index)):
                 raise IndexingError("Handle elsewhere")
```

Comment:
is line 978 still needed / hit?

Reply:
are you referencing this line?
`if len(tup) > self.obj.ndim:
            raise IndexingError("Too many indexers. handle elsewhere")`
I have added the test case where it hits.